entrypoint.sh

```diff
@@ -1,42 +1,37 @@
 #!/bin/bash
-set -e
-set -o pipefail
+set -eo pipefail
 
 main() {
-  echo "Current ref: ${GITHUB_REF}"
+  PR_REF="${GITHUB_REF%/merge}/head"
+  BASE_REF="${GITHUB_BASE_REF}"
 
-  if [[ "$GITHUB_REF" != "refs/heads/"* ]]; then
-    echo "Only check branches, not tags or other refs."
-    exit 0
-  fi
-
-  BRANCH=${GITHUB_REF:11}
-  echo "Current branch: ${BRANCH}"
+  echo "Current ref: ${PR_REF}"
+  echo "Base ref: ${BASE_REF}"
 
-  if [ "$BRANCH" == "master" ]; then
-    echo "No check of master branch needed."
-    exit 0
+  if [[ "${PR_REF}" != "refs/pull/"* ]]; then
+    echo "This check works only with pull_request events"
+    exit 1
   fi
 
   # Using git directly because the $GITHUB_EVENT_PATH file only shows commits in
   # most recent push.
-  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin master
-  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --shallow-exclude=master origin ${BRANCH}
+  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin "${BASE_REF}:__ci_base"
+  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --shallow-exclude="${BASE_REF}" origin "${PR_REF}:__ci_pr"
   # Get the list before the "|| true" to fail the script when the git cmd fails.
-  COMMIT_LIST=`/usr/bin/git log --pretty=format:%s origin/master..origin/${BRANCH}`
+  COMMIT_LIST=$(/usr/bin/git log --pretty=format:%s __ci_base..__ci_pr)
 
-  FIXUP_COUNT=`echo $COMMIT_LIST | grep fixup! | wc -l || true`
+  FIXUP_COUNT=$(echo ${COMMIT_LIST} | grep 'fixup!' | wc -l || true)
```

Comment:
This reformatting should be a separate PR, separate commit, or not done since it's cluttering the review of the important changes. Are you using a bash-script automatic formatter? What standards are you changing this formatting to?

Reply:
Sorry. I'm used to doing stuff like this on the fly since most things I contribute to don't keep their git history clean. Of course that's not the case here 😆

I've updated the PR to keep the formatting as-is. I recommend running shellcheck over the script though, I think backticks are often discouraged and you're also not consistent about `$VAR` vs `${VAR}`. I wasn't going for some specific standard, just more consistency and what I had in my head as best practices.

Reply:
Good point about the consistency. This works for now though.